pkg/version/version_info.go

```diff
@@ -0,0 +1,67 @@
+package version
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+)
+
+type versionInfo struct {
+	major   int
+	minor   int
+	release int
+}
+
+var versionRegex = regexp.MustCompile(`([\d]+)\.([\d]+)\.([\d]+)`)
+
+// compareVersionInfo returns:
+// 	0: if a == b
+//  n > 0: if a > b
+//  n < 0: if a < b
+//  0 with error: if a == nil || b == nil
+func compareVersionInfo(a *versionInfo, b *versionInfo) (int, error) {
+	if a == nil || b == nil {
+		return 0, fmt.Errorf("parameter must not be null")
+	}
+
+	// Check major version
+	result := a.major - b.major
+	if result != 0 {
+		return result, nil
+	}
+
+	// Major is equal, check minor
+	result = a.minor - b.minor
+	if result != 0 {
+		return result, nil
+	}
+
+	// Major and minor is equal, check release
+	result = a.release - b.release
+	return result, nil
+}
+
+func extractVersion(versionString string) (*versionInfo, error) {
+	version := versionRegex.FindStringSubmatch(versionString)
+
+	if len(version) < 4 {
+		return nil, fmt.Errorf("version malformed: %s", versionString)
```

Comment:
Just in case, things such as `versionRegex.FindStringSubmatch("asdadasdsd.asd1.2.3")` would pass the validation, although they most certainly won't happen.